tests/test_gear_generator.py

```diff
@@ -0,0 +1,78 @@
+import cadquery as cq
+from plugins.gear_generator.gear_generator import *
+from unittest import TestCase
+from math import sin, radians
+
+class TestGearGenerator(TestCase):
+    def test_bevel_gear(self):
+        """
+        Tests if the bevel gear has been created successfully 
+        by checking if it's valid and it's resulting volume
+        Numerical values volumes were obtained with the Volume() functions on a Windows 10 machine with python 3.9.2
+        """
+        m = 1.5
+        z = 16
+        b = 6        
+        delta = 45
+        R = (m*z/2) / sin(radians(delta))
+        alpha = 20
+        helix_angle = 20
+        clearance = 6
+
+        gear1 = BevelGear(m, z, b, delta, R, alpha = alpha, clearance = clearance).build()
+        gear2 = BevelGear(m, z, b, delta, R, alpha = alpha, clearance = clearance, helix_angle=helix_angle).build()
+    
+        self.assertTrue(gear1.val().isValid())
+        self.assertAlmostEqual(gear1.val().Volume(),2505.7966225157024)
+
+        self.assertTrue(gear2.val().isValid())
+        self.assertAlmostEqual(gear1.val().Volume(),2505.7531046471067)
+
+
+    def test_bevel_gear_system(self):
+        """
+        Tests if the bevel gear system been created successfully 
+        by checking if it's valid and it's resulting volume
+        Numerical values volumes were obtained with the Volume() functions on a Windows 10 machine with python 3.9.2
+        """
+        m = 1
+        z1 = 16
+        z2 = 22
+        b = 2
+        alpha = 20
+        helix_angle = 20
+        clearance = 3
+
+        pinion1, gear1 = BevelGearSystem(m, z1, z2, b, clearance1 = clearance, clearance2= clearance, alpha=alpha).build()
+        self.assertTrue(pinion1.val().isValid())
+        self.assertTrue(gear1.val().isValid())
+        self.assertAlmostEqual(pinion1.val().Volume(),483.3443067836134)
+        self.assertAlmostEqual(gear1.val().Volume(),1126.5921554741049)
+
+        pinion2, gear2 = BevelGearSystem(m, z1, z2, b, clearance1 = clearance, clearance2= clearance, helix_angle=helix_angle, alpha=alpha).build()
+        self.assertTrue(pinion2.val().isValid())
+        self.assertTrue(gear2.val().isValid())
+        self.assertAlmostEqual(pinion2.val().Volume(),483.3523144907097)
+        self.assertAlmostEqual(gear2.val().Volume(),1126.605502545581)
+
+    def test_gear(self):
+        """
+        Tests if the gear has been created successfully 
+        by checking if it's valid and it's resulting volume
+        Numerical values volumes were obtained with the Volume() functions on a Windows 10 machine with python 3.9.2
```

Comment:
Assert result on Ubuntu 18.04. This one is more concerning to me, like maybe invalid geometry is being generated on Linux.
```
>       self.assertAlmostEqual(gear1.val().Volume(),10113.473492717025)
E       AssertionError: 10045.87763052491 != 10113.473492717025 within 7 places (67.59586219211451 difference)
```

Reply:
The resulting gear is a valid solid in Ubuntu 18.04. I even exported it to STL and sliced it to make sure. A discrepancy of 67.6 seems like more than rounding error to me, but it's not due to the shape being invalid.

![Screenshot from 2021-04-30 13-11-02](https://user-images.githubusercontent.com/1015439/116729761-8de98c00-a9b5-11eb-8128-3ea2bf0572ac.png)

Reply:
That's 0.7 % difference, it seems quite high though. But if it end up looking as expected I don't know what we should do, should we not test the validity by volume ? I took this testing method from the cq tests, maybe you have idea of better testing ?